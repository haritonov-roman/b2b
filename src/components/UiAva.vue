<script>
export default {
  name: 'UiAva',

  props: {
    src: {
      type: String,
      required: true,
      default: ''
    }
  }
}
</script>

<template>
  <img
    :src="src"
    class="ava"
    alt="ava"
  />
</template>

<style lang="scss" scoped>
@import "@central-design-system/components/dist/mixins/scss/layout/convert";

.ava {
  width: toRem(40px);
  height: toRem(40px);
  border-radius: 50%;
  overflow: hidden;
}
</style>
