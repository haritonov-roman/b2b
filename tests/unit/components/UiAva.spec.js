import { shallowMount } from '@vue/test-utils'
import UiAva from '@/components/UiAva'

describe('UiAva.vue', () => {
  const src = 'https://avatars.githubusercontent.com/u/46739861?u=b461863a46253deb150f48f0ae61823f763c0475&v=4'

  const wrapper = shallowMount(UiAva, {
    propsData: { src }
  })

  it('Проверка рендеринга', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('Проверка входящих данных', () => {
    expect(wrapper.attributes('src')).toMatch(/(^https?:\/\/)?/i)
  })

  it('Снимок', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
