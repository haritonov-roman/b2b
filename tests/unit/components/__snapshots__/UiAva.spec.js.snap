// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UiAva.vue Снимок 1`] = `
<img
  alt="ava"
  class="ava"
  src="https://avatars.githubusercontent.com/u/46739861?u=b461863a46253deb150f48f0ae61823f763c0475&v=4"
/>
`;
